Mathlib/RingTheory/OrzechProperty.lean

```diff
@@ -0,0 +1,95 @@
+/-
+Copyright (c) 2024 Jz Pan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jz Pan
+-/
+import Mathlib.RingTheory.Finiteness
+import Mathlib.Logic.Equiv.TransferInstance
+
+/-!
+
+# Orzech property of rings
+
+In this file we define the following property of rings:
+
+- `OrzechProperty R` is a type class stating that `R` satisfies the following property:
+  for any finitely generated `R`-module `M`, any surjective homomorphism `f : N → M`
+  from a submodule `N` of `M` to `M` is injective.
+  It was introduced in papers by Orzech [orzech1971], Djoković [djokovic1973] and
+  Ribenboim [ribenboim1971], under the names `Π`-ring or `Π₁`-ring.
+  It implies the strong rank condition (that is, the existence of an injective linear map
+  `(Fin n → R) →ₗ[R] (Fin m → R)` implies `n ≤ m`)
+  if the ring is nontrivial (see `Mathlib/LinearAlgebra/InvariantBasisNumber.lean`).
+
+It's proved in the above papers that
+
+- a Noetherian ring (not necessarily commutative) satisfies the `OrzechProperty`
```

Comment:
What is a Noetherian non-commutative ring? Is this left Noetherian, right Noetherian, or both? Sorry to be so ignorant about noncommutative rings!

Reply:
It's `IsNoetherianRing`, which I think is left Noetherian, i.e. `R` itself is a Noetherian left `R`-module.